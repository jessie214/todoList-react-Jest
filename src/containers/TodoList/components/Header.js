import React, { Component } from 'react';
import '../style.css';

class Header extends Component {
  constructor(props) {
    super(props);
    this.state = {
      value:''
    }
  }

 handleInputChange = (e) => {
    this.setState({
      value:e.target.value
    })    
  }

  handleInputKeyUp = (e) => {
    if (e.keyCode === 13 && this.state.value) {
      this.props.addUndoItem(this.state.value)
      this.setState({value:''})
    }
  }

  render() {
    return (
      <div className='header'>
        <div className='hader-content'>
        TodoList
          <input data-test="input"
            placeholder='addtodo'
            className='header-input'
            value={this.state.value}
            onChange={this.handleInputChange}
            onKeyUp={this.handleInputKeyUp}
        /> 
      </div>               
    </div>
    )
  }
}
// const Header = (props) => {
//   const [value, setValue] = useState('');
//   const handleInputChange = (e) => {
//     if (e) {
//       setValue(e.target.value)
//     console.log(e.target.value)
//     }
    
//   }
//   const handleInputKeyUp = (e) => {
//     if (e.keyCode === 13 && value) {
//       props.addUndoItem(value)
//     }
//   }
//   return (
//     <div>
//       <input
//         data-test="input"
//         className='searchInput'
//         value={value}
//         onChange={(e) => handleInputChange(e)}
//         onKeyUp={handleInputKeyUp}
//       />
//     </div>
//   )
// }

export default Header;