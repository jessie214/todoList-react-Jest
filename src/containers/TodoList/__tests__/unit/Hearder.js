import React from 'react';
import {shallow} from 'enzyme';
// import Adapter from 'enzyme-adapter-react-16';
import Header from '../../components/Header';
import { findTestWrapper } from '../../../../utils/testUtils';

// Enzyme.configure({ adapter: new Adapter() });
describe("Header component", () => {
  it('style is normal', () => {
    const wrapper = shallow(<Header />); 
    expect(wrapper).toMatchSnapshot();
  });

  it('It include a input', () => {
    const wrapper = shallow(<Header />);
    const inputElem = findTestWrapper(wrapper, 'input');
    expect(inputElem.length).toBe(1);
  });

  it('The initialization input should be empty', () => {
    const wrapper = shallow(<Header />);
    const inputElem = findTestWrapper(wrapper, 'input');
    expect(inputElem.prop('value')).toEqual('');
  });
  
  it('The input changes as the user enters content', () => {
    const wrapper = shallow(<Header />);
    const inputElem = findTestWrapper(wrapper, 'input');
    inputElem.simulate('change', { target: { name: 'email', value: 50 } });
    wrapper.update();
    expect(
      wrapper
        .find("[data-test='input']")
        .props().value,
    ).toEqual(50);
  });

  it('If there is no input content when press Enter, there is no action', () => {
    const fn = jest.fn();
    const wrapper = shallow(<Header addUndoItem={fn} />);
    wrapper.setState({value:''})
    findTestWrapper(wrapper, 'input').simulate('keyUp', {
      keyCode: 13
    });
    expect(fn).not.toHaveBeenCalled();
  });
  
  it('called when there is content after press Enter', () => {
    const fn = jest.fn();
    const wrapper = shallow(<Header addUndoItem={fn} />);
    const userInput ='learn react'
    wrapper.setState({value:userInput})
    findTestWrapper(wrapper, 'input').simulate('keyUp', {
      keyCode: 13
    });
    expect(fn).toHaveBeenCalled();
    expect(fn).toHaveBeenLastCalledWith(userInput);
  
  });
  
  it('the content of input should be cleared after press Enter', () => {
    const fn = jest.fn();
    const wrapper = shallow(<Header addUndoItem={fn} />);
    const inputElem = findTestWrapper(wrapper, 'input');
    const userInput ='learn react'
    wrapper.setState({value:userInput})
    wrapper.find("[data-test='input']").simulate('keyUp', {
      keyCode: 13
    });
    const newInputelem = findTestWrapper(wrapper, 'input');
    expect(newInputelem.prop('value')).toBe('')
  });  

})








