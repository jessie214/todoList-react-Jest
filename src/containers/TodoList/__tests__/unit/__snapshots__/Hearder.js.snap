// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The header rendering style is normal 1`] = `
<div
  className="header"
>
  <div
    className="hader-content"
  >
    TodoList
    <input
      className="header-input"
      data-test="input"
      onChange={[Function]}
      onKeyUp={[Function]}
      placeholder="addtodo"
      value=""
    />
  </div>
</div>
`;
